bs/src/tablecloth.ml

```diff
@@ -432,28 +432,46 @@ module Tuple2 = struct
   
   let second ((_, b) : 'a * 'b) : 'b = b
 
+  let map ~(f : 'b -> 'y) ((a, b) : 'a * 'b) : 'a * 'y = (a, f b)
+
   let mapFirst ~(f : 'a -> 'x) ((a, b) : 'a * 'b) : 'x * 'b = (f a, b)
 
   let map_first = mapFirst
 
-  let mapSecond ~(f : 'b -> 'y) ((a, b) : 'a * 'b) : 'a * 'y = (a, f b)
+  let mapSecond = map
 
   let map_second = mapSecond
 
-  let mapBoth ~(f : 'a -> 'x) ~(g : 'b -> 'y) ((a, b) : 'a * 'b) : 'x * 'y = (f a, g b)
+  let mapEach ~(f : 'a -> 'x) ~(g : 'b -> 'y) ((a, b) : 'a * 'b) : 'x * 'y = (f a, g b)
+
+  let map_each = mapEach
+  
+  let mapAll ~(f : 'a -> 'b) (a1, a2) = (f a1, f a2)
+
+  let map_all = mapAll
+
+  let swap (a, b) = (b, a)
 
-  let map_both = mapBoth
+  let toList (a, b) = [a; b]
+
+  let to_list = toList
 end
 
 module Tuple3 = struct
   let create a b c = (a, b, c)
 
   let first ((a, _, _) : 'a * 'b * 'c) : 'a = a
 
-  let second ((_, b, _) : 'a * 'b *'c) : 'b = b
+  let second ((_, b, _) : 'a * 'b * 'c) : 'b = b
   
   let third ((_, _, c) : 'a * 'b * 'c) : 'c = c
 
+  let head ((a, b, _) : 'a * 'b * 'c): ('a * 'b) = (a, b)
```

Comment:
The word `init` is a known word for the opposite of `tail`, so I'd use that. (`tail` is everything but the first, works with `head`, `init` is everything but the last, works with `last`).

Reply:
I will un-rename it. Its not something I had come across before.